src/railfence.rs

```diff
@@ -0,0 +1,284 @@
+            // the distance from the end of the cycle.
+            let row = if i % cycle <= cycle / 2 {
+                i % cycle
+            }
+            else {
+                cycle - i%cycle
+            };
+
+            // Fill cell with an arbitrary letter
+            table[row][col] = (true, 'a');
+        }
+
+        // Fill the identified positions in the table with the ciphertext, line by line
+        let mut ct_iter = cipher_text.chars();
+        for row in table.iter_mut() {
+            for entry in row.iter_mut() {
+                if entry.0 {
+                    *entry = (true, ct_iter.next().unwrap());
+                }
+            }
+        }
+
+        // Read the plaintext in a zigzag
+        let mut message = String::new();
+        for i in 0..cipher_text.len() {
+            let col = i;
+            let cycle = 2 * self.key - 2;
+            let row = if i % cycle <= cycle / 2 {
+                i % cycle
+            }
+            else {
+                cycle - i%cycle
+            };
+
+            message.push(table[row][col].1);
+        }
+
+        message
+    }
+
+}
+
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+
+    #[test]
+    fn encrypt_test() {
+        let key = 6;
+        let message = "attackatdawn";
+        let r = Railfence::new(key).unwrap();
+        assert_eq!("awtantdatcak", r.encrypt(message));
+    }
+
+    #[test]
+    fn encrypt_mixed_case() {
+        let key = 3;
+        let message = "Hello, World!";
+        let r = Railfence::new(key).unwrap();
+        assert_eq!("Hoo!el,Wrdl l", r.encrypt(message));
+    }
+
+    #[test]
+    fn encrypt_short_key() {
+        let key = 1;
+        let message = "attackatdawn";
+        let r = Railfence::new(key).unwrap();
+        assert_eq!("attackatdawn", r.encrypt(message));
+    }
+
+    #[test]
+    fn encrypt_long_key() {
+        let key = 20;
+        let message = "attackatdawn";
+        let r = Railfence::new(key).unwrap();
+        assert_eq!("attackatdawn", r.encrypt(message));
+    }
+
+    #[test]
+    fn decrypt_test() {
+        let key = 6;
+        let message = "awtantdatcak";
+        let r = Railfence::new(key).unwrap();
+        assert_eq!("attackatdawn", r.decrypt(message));
+    }
+
+    #[test]
+    fn decrypt_short_key() {
+        let key = 1;
+        let message = "attackatdawn";
+        let r = Railfence::new(key).unwrap();
+        assert_eq!("attackatdawn", r.decrypt(message));
+    }
+
+    #[test]
+    fn decrypt_mixed_case() {
+        let key = 3;
+        let message = "Hoo!el,Wrdl l";
+        let r = Railfence::new(key).unwrap();
+        assert_eq!("Hello, World!", r.decrypt(message));
+    }
+
+    #[test]
+    fn decrypt_long_key() {
+        let key = 20;
+        let message = "attackatdawn";
+        let r = Railfence::new(key).unwrap();
+        assert_eq!("attackatdawn", r.decrypt(message));
+    }
+
+    #[test]
+    #[should_panic]
```

Comment:
Remove the `#[should_panic]` and instead `assert!(Railfence::new(0).is_err())`. So its standard with the other tests